flexget/plugins/metainfo/nfo_lookup.py

```diff
@@ -0,0 +1,241 @@
+                if values is not None:
+                    d[nfo_field_name] = values
+
+        return d
+
+    @staticmethod
+    def _extract_single_field(root, name):
+        """
+        Use this method to get fields from the root XML tree that only appear once,
+        such as 'title', 'year', etc.
+        """
+        f = root.find(name)
+        if f is not None:
+            return f.text
+        else:
+            return None
+
+    @staticmethod
+    def _extract_multiple_field(root, name):
+        """
+        Use this method to get fields from the root XML tree that can appear more
+        than once, such as 'actor', 'genre', 'director', etc.
+        """
+        if name == 'actor':
+            values = []
+        else:
+            values = [i.text for i in root.findall(name)]
+
+        if len(values) > 0:
+            return values
+        else:
+            return None
+
+    def add_imdb_fields_to_dict(self, fields):
+        """
+        Take the dictionary of 'nfo fields' and add extra keys with 'imdb fields'
+        corresponding to the 'nfo fields'
+        """
+        single_field_mapping = {
+            # u'nfo_id': u'imdb_id',
+            u'nfo_title': u'imdb_name',
+            u'nfo_originaltitle': u'imdb_original_name',
+            # u'nfo_votes' u'imdb_votes',
+            # u'nfo_year': u'imdb_year',
+            u'nfo_plot': u'imdb_plot_outline',
+            # u'nfo_rating': u'imdb_score',
+            # u'nfo_thumb': u'imdb_photo'
+        }
+
+        # multiple_field_mapping = {
+        #     # List of imdb genres
+        #     u'nfo_genre': u'imdb_genres',
+        #     # Directors dictionary (imdbid, name)
+        #     u'nfo_director': u'imdb_directors',
+        #     # Actors dictionary (key: imdbid, value: name)
+        #     u'nfo_actor': u'imdb_actors',
+        # }
+
+        # extra_imdb_fields = ['imdb_url',
+        #                      # 'imdb_photo',
+        #                      'imdb_languages']
+
+        # Add imdb entries with single value
+        for nfo_name, imdb_name in single_field_mapping.items():
+            if nfo_name in fields:
+                fields[imdb_name] = fields[nfo_name]
+
+        # Add imdb fields with multiple values
+        if 'nfo_genre' in fields:
+            fields[u'imdb_genres'] = fields[u'nfo_genre']
+
+        # The nfo file can also have actors and directors, but there is no gain
+        # replacing the the values we got from the IMDB lookup.
+
+    def get_nfo_filename(self, entry):
+        location = entry.get('location')
+        # TODO: maybe create an option to set a different extension than just .nfo
+        nfo_full_filename = os.path.splitext(location)[0] + '.nfo'
+
+        if os.path.isfile(nfo_full_filename):
+            return nfo_full_filename
+        else:
+            return None
+
+    @property
+    def movie_identifier(self):
+        """Returns the plugin main identifier type"""
```

Comment:
I'm actually new to flexget and looking at the source I saw some plugins defining this property. It could be something important that I didn't see in the documentation and that's why I have included. I will remove this then.